lib/scanners/base.js

```diff
@@ -34,17 +34,41 @@ var BaseScanner = function (beaconMapper, beaconRegionMapper, beaconRegionMonito
     this.init = function (networkIdentifier) {
         if (!_.isUndefined(networkIdentifier)) {
             self.networkId = networkIdentifier;
+            var hasPermissions = function () {
```

Comment:
Great minds think alike ;)
Already explained the naming issue and rewrote the whole thing.